circuits/cpp/src/aztec3/circuits/kernel/private/native_private_kernel_circuit.test.cpp

```diff
@@ -0,0 +1,498 @@
+#include "c_bind.h"
+#include "testing_harness.hpp"
+
+#include "aztec3/circuits/apps/test_apps/basic_contract_deployment/basic_contract_deployment.hpp"
+#include "aztec3/circuits/apps/test_apps/escrow/deposit.hpp"
+#include "aztec3/utils/circuit_errors.hpp"
+
+#include "barretenberg/numeric/random/engine.hpp"
+#include "barretenberg/serialize/test_helper.hpp"
+#include <barretenberg/common/test.hpp>
+
+#include <gtest/gtest.h>
+
+#include <cstdint>
+
+namespace {
+
+using aztec3::circuits::apps::test_apps::basic_contract_deployment::constructor;
+using aztec3::circuits::apps::test_apps::escrow::deposit;
+
+using aztec3::circuits::kernel::private_kernel::testing_harness::do_private_call_get_kernel_inputs_init;
+using aztec3::circuits::kernel::private_kernel::testing_harness::do_private_call_get_kernel_inputs_inner;
+using aztec3::circuits::kernel::private_kernel::testing_harness::validate_deployed_contract_address;
+using aztec3::circuits::kernel::private_kernel::testing_harness::validate_no_new_deployed_contract;
+using aztec3::utils::CircuitErrorCode;
+
+}  // namespace
+
+namespace aztec3::circuits::kernel::private_kernel {
+
+/**
+ **************************************************************
+ * Native initial private kernel circuit tests.
+ **************************************************************
+ */
+
+/**
+ * @brief Some private circuit simulation (`deposit`, in this case)
+ */
+TEST(native_private_kernel_init_tests, deposit)
+{
+    NT::fr const amount = 5;
+    NT::fr const asset_id = 1;
+    NT::fr const memo = 999;
+
+    auto const& private_inputs = do_private_call_get_kernel_inputs_init(false, deposit, { amount, asset_id, memo });
+    DummyComposer composer = DummyComposer("private_kernel_tests__native_deposit");
+    auto const& public_inputs = native_private_kernel_circuit_initial(composer, private_inputs);
+
+    EXPECT_TRUE(validate_no_new_deployed_contract(public_inputs));
+
+    // Check the first nullifier is hash of the signed tx request
+    ASSERT_EQ(public_inputs.end.new_nullifiers[0], private_inputs.signed_tx_request.hash());
+}
+
+/**
+ * @brief Some private circuit simulation (`constructor`, in this case)
+ */
+TEST(native_private_kernel_init_tests, basic_contract_deployment)
+{
+    NT::fr const arg0 = 5;
+    NT::fr const arg1 = 1;
+    NT::fr const arg2 = 999;
+
+    auto const& private_inputs = do_private_call_get_kernel_inputs_init(true, constructor, { arg0, arg1, arg2 });
+    DummyComposer composer = DummyComposer("private_kernel_tests__native_basic_contract_deployment");
+    auto const& public_inputs = native_private_kernel_circuit_initial(composer, private_inputs);
+
+    EXPECT_TRUE(validate_deployed_contract_address(private_inputs, public_inputs));
+}
```

Comment:
Can we assert here that dummy composer has no failures? If it does we should print them out:
```cpp
    auto failure = composer.get_first_failure();
    if (failure.code != CircuitErrorCode::NO_ERROR) {
        info("failure: ", failure);
    }
    ASSERT_FALSE(composer.failed());
```

Reply:
Right, I missed adding it. Will do.

Reply:
Done.